### PR TITLE
Richard/Squarelotron

### PR DESCRIPTION
**Pull request by RichardSeverich**

### CODE REVIEW

File: src/main/java/org/minions/devfund/richard/square/Squarelotron.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
+package org.minions.devfund.richard.square;
+
+/**
+ *
+ */
+public class Squarelotron {
+
+    private int[][] squarelotron;
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/richard/square/Squarelotron.java]: MAJOR Rename field "squarelotron" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1700)

**Comment by carledriss** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by RichardSeverich** [comment]
image

File: src/test/java/org/minions/devfund/richard/SquareTest.java [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+     *
+     */
+    @Test
+    public void testUpsideDownFlipMethod() {
+        Square mySquare = new Square(FOUR);
+        final String expectedValue = "13 14 15 16 "
+                + "9 6 7 12 "
+                + "5 10 11 8 "
+                + "1 2 3 4 ";
+        assertEquals(expectedValue, mySquare.upsideDownFlip(ONE).print());
```
Comment (carledriss) [inline src/test/java/org/minions/devfund/richard/SquareTest.java:43]: Please Consider using arrays for the assertions and not use the print method.

File: src/test/java/org/minions/devfund/richard/SquareTest.java [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+     *
+     */
+    @Test
+    public void testMainDiagonalFlipMethod() {
+        Square mySquare = new Square(FOUR);
+        final String expectedValue = "1 5 9 13 "
+                + "2 6 7 14 "
+                + "3 10 11 15 "
+                + "4 8 12 16 ";
+        assertEquals(expectedValue, mySquare.mainDiagonalFlip(ONE).print());
```
Comment (carledriss) [inline src/test/java/org/minions/devfund/richard/SquareTest.java:56]: same as above

File: src/test/java/org/minions/devfund/richard/SquareTest.java [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+     *
+     */
+    @Test
+    public void testRotateRightMethod() {
+        Square mySquare = new Square(FOUR);
+        final String expectedValue = "13 9 5 1 "
+                + "14 10 6 2 "
+                + "15 11 7 3 "
+                + "16 12 8 4 ";
+        assertEquals(expectedValue, mySquare.rotateRight(ONE).print());
```
Comment (carledriss) [inline src/test/java/org/minions/devfund/richard/SquareTest.java:69]: same as above

**Review by carledriss — CHANGES_REQUESTED** [review]

File: src/test/java/org/minions/devfund/richard/SquareTest.java (hunk 2 above)
Comment (RichardSeverich) [reply src/test/java/org/minions/devfund/richard/SquareTest.java:43]: I used array in other tests, this test is specific for print method.

**Review by carledriss — APPROVED** [review]

**Pull request merged**